doc/specs/stdlib_string_type.md

```diff
@@ -0,0 +1,1607 @@
+
+#### Result value
+
+The result is a scalar character value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+character(len=:), allocatable :: dlc
+
+string = "Character sequence"
+dlc = char(string)
+! dlc == "Character sequence"
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Char function (position variant)
+
+#### Description
+
+Return the character sequence represented by the string.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):char(interface)]] (string, pos)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Instance of a `string_type`. This argument is `intent(in)`.
+- `pos`: Integer of default type. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a scalar character value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+character(len=:), allocatable :: dlc
+character(len=1), allocatable :: chars(:)
+
+string = "Character sequence"
+dlc = char(string, 3)
+! dlc == "a"
+chars = char(string, [3, 5, 8, 12, 14, 15, 18])
+! chars == ["a", "a", "e", "e", "u", "e", "e"]
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Char function (range variant)
+
+#### Description
+
+Return the character sequence represented by the string.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):char(interface)]] (string, start, last)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Pure function.
+
+#### Argument
+
+- `string`: Instance of a `string_type`. This argument is `intent(in)`.
+- `start`: Integer of default type. This argument is `intent(in)`.
+- `last`: Integer of default type. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a scalar character value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+character(len=:), allocatable :: dlc
+
+string = "Fortran"
+dlc = char(string, 1, 4)
+! dlc == "Fort"
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Ichar function
+
+Character-to-integer conversion function.
+
+#### Description
+
+Returns the code for the character in the first character position of the
+character sequence in the system's native character set.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):ichar(interface)]] (string)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Instance of a `string_type`. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: code
+
+string = "Fortran"
+code = ichar(string)
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Iachar function
+
+Code in ASCII collating sequence.
```

Comment:
I suggest to remove this line.
```suggestion
```

Reply:
How come? The other functions also have a short one line description.

Reply:
> How come? The other functions also have a short one line description.

Well, I would remove all of them. There is a section "description".Therefore, I didn't understand why it was described before the section "description".